src/common/options/rgw.yaml.in

```diff
@@ -2864,12 +2864,45 @@ options:
   with_legacy: true
 # extra keys that may be used for aws:kms
 # defined as map "key1=YmluCmJvb3N0CmJvb3N0LQ== key2=b3V0CnNyYwpUZXN0aW5nCg=="
+- name: rgw_crypt_s3_kms_cache_enabled
+  type: bool
+  level: advanced
+  desc: Enable SSE-KMS encryption key cache.
+  default: false
+  with_legacy: true
+  services:
+  - rgw
+- name: rgw_crypt_s3_kms_cache_max_size
+  type: uint
+  level: advanced
+  desc: Maximum number of SSE-KMS secrets cached.
+    Each key needs 32 byte (AES-256) + overhead in memory
+  default: 128
+  with_legacy: true
+  services:
+  - rgw
+- name: rgw_crypt_s3_kms_cache_ttl
+  type: uint
+  level: advanced
+  desc: Time in seconds after which the KMS cache evicts entries. Set to 0 to disable.
```

Comment:
if ttl is a security requirement, we should probably not allow it to be disabled or raised above some reasonable maximum. this yaml can enforce min/max values, ex
```yaml
  min: 10
  max: 3600
```